examples/widgets/slider_snap_demo.py

```diff
@@ -0,0 +1,62 @@
+"""
+==========================
+Slider Value Snapping Demo
+==========================
+
+You can snap slider values to a discrete values using either the ``valstep``
+argument.
+
+In this example the Freq slider is constrained to be multiples of pi, and the
+Amp slider uses an array as the ``valstep`` argument to more densely sample
+the first part of its range.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.widgets import Slider, Button
+
+fig, ax = plt.subplots()
+plt.subplots_adjust(bottom=0.25)
+t = np.arange(0.0, 1.0, 0.001)
+a0 = 5
+f0 = 3
+s = a0 * np.sin(2 * np.pi * f0 * t)
+l, = plt.plot(t, s, lw=2)
+
+axcolor = 'lightgoldenrodyellow'
+axfreq = plt.axes([0.25, 0.1, 0.65, 0.03], facecolor=axcolor)
+axamp = plt.axes([0.25, 0.15, 0.65, 0.03], facecolor=axcolor)
+
+# define the values to use for snapping
+delta_f = np.pi
+allowed_amplitudes = np.concatenate([np.linspace(.1, 5, 100), [7, 8, 9, 10]])
```

Comment:
```suggestion
allowed_amplitudes = np.concatenate([np.linspace(.1, 5, 100), [6, 7, 8, 9]])
```
otherwise there's an unexpected gap at 6.